wallet/src/lib.did

```diff
@@ -164,6 +195,11 @@ service : {
   get_events: (opt record { from: opt nat32; to: opt nat32; }) -> (vec Event) query;
   get_chart: (opt record { count: opt nat32; precision: opt nat64; } ) -> (vec record { nat64; nat64; }) query;
 
+  // Children
+  list_children: () -> (vec ChildInfo) query;
+  get_child_events: (principal, opt record { from: opt nat32; to: opt nat32; }) -> (opt vec ChildEvent) query;
```

Comment:
A convention in use here and in other canisters (example: [certified assets](https://github.com/dfinity/certified-assets/blob/main/assets.did)) is to use a single record as the parameter.  This allows for adding new parameters later (for example, here, that might be pagination in the future) without breaking the interface.
```suggestion
  get_child_events: (record { id: principal, from: opt nat32; to: opt nat32; }) -> (opt vec ChildEvent) query;
```

Reply:
It does specify something meaningful, that there was no known child canister by that principal. And should that be followed for convention, or for principle? Because the Candid specification says additional optional function parameters are not breaking changes, unless you're referring to the Rust interface, in which case adding new fields to a struct is also a breaking change. But if it's just to maintain convention then no reason not to have it.

Reply:
It does specify something meaningful, that there was no such child canister. And is that followed just for convention, or for principle? Because the Candid docs say new optional function parameters can be added without it being a breaking Candid change, unless you mean breaking Rust change, in which case adding fields to a struct is also a breaking Rust change.

Reply:
Consider the following service .did definition:
```
service : {
    "greet": (text) -> (text) query;
    "greet_with_opt_param": (text, opt text) -> (text) query;

    "record_greet": (record { a: text }) -> (text) query;
    "record_greet_with_opt_param": (record { a: text; b: opt text }) -> (text) query;
}

All of the functions are adapted from `dfx new --type rust <project>` to return a string that includes all of the passed arguments.

```
Suppose "greet" and "record_greet" are the initial versions of some function. Later, it's desired to add a parameter in a way that doesn't break the interface ("greet_with_opt_param" and "record_greet_with_opt_param").

We can call the first versions just fine, for example with dfx:
```
$ dfx canister call compat greet '("A")'
("Hello, A!")
$ dfx canister call compat record_greet '(record{a="abc"})'
("Hello, abc!")
```

Now, let's suppose an upgrade adds a new optional parameter.  Let's see if the change is non-breaking:
```
$ dfx canister call compat greet_with_opt_param '("A")'
Invalid data: Unable to serialize Candid values: wrong number of argument values
$ dfx canister call compat record_greet_with_opt_param '(record{a="abc"})'
("Hello, abc! None")
```

We were able to add a new parameter without breaking the interface, if the parameter was a single record argument.  But adding a new argument breaks invocations that used to work.

This is why we use this convention in the assets canister ([example](https://github.com/dfinity/certified-assets/blob/main/assets.did#L113)) and why I am suggesting that we do the same here.

Reply:
Ah, I didn't realize dfx checked exact parameter count instead of allowing function subtypes. The rule makes sense then.